lib/dry/system/provider_sources/settings/config.rb

```diff
@@ -41,7 +41,11 @@ def self.load(root:, env:, loader: Loader)
                 value = loader[setting_name.to_s.upcase]
 
                 begin
-                  settings_obj.config.public_send(:"#{setting_name}=", value) if value
+                  if value
+                    settings_obj.config.public_send(:"#{setting_name}=", value)
+                  else
+                    settings_obj.config[setting_name]
+                  end
```

Comment:
Basically, if the loaded doesn't find a value, we still evaluate the setting to make sure the constructor is happy with the setting being `nil`.